src/plugins/intel_cpu/src/nodes/tensoriterator.cpp

```diff
@@ -303,24 +338,49 @@ void DynamicBuffer::move_data() {
 
     copy(reinterpret_cast<const uint8_t*>(from->GetPtr()), get_ptr(*mem_holder_buffer.get()) + chunk_offset_in_byte,
          src_stride, dst_stride, count, src_stride);
+
+    // adjust for next execution
+    num_iters++;
+    if (map_rule.stride > 0.0f) {
+        chunk_offset_in_byte += mem_holder_unit;
+    } else {
+        chunk_offset_in_byte -= mem_holder_unit;
+    }
 }
 
 void DynamicBuffer::transfer(const Node* node) {
     if (mem_holder_buffer) {
+        const auto axis = map_rule.axis;
+        const auto stride = map_rule.stride;
+        const auto abs_stride = std::abs(stride);
+
+        const auto mem_holder_desc = mem_holder_buffer->get_desc();
+        const auto mem_holder_dims = mem_holder_desc.dims();
+        auto dims = mem_holder_dims;
+
+        dims[axis] = abs_stride * num_iters;
         const auto desc = node->getBaseMemDescAtOutputPort(map_rule.from)->cloneWithNewDims(
-                DnnlExtensionUtils::convertToVectorDims(mem_holder_buffer->get_desc().dims()));
-        redefineToMemories(to, desc);
+                DnnlExtensionUtils::convertToVectorDims(dims));
 
-        copy(get_ptr(*mem_holder_buffer.get()), reinterpret_cast<uint8_t*>(to.front()->GetPtr()), 0, 0, 1, to.front()->GetSize());
+        if (dims[axis] == mem_holder_dims[axis]) {
+            // straightly share mem_holder_buffer to outside graph
+            redefineToMemories(to, desc, mem_holder_buffer->get_data_handle());
```

Comment:
> The lifetime of the `to` memory objects is longer than the lifetime of the `mem_holder_buffer`, which may lead to the situation when the `data_handle` pointer is expired but the memory objects steel reference it. 
My idea is "to" memory objects will take over the ownership of  data_handle of "mem_holder_buffer" since then. Will it expire if it's taken?

> So Does it really worth the cost of one extra copy? What do you think?
Do you mean keep the copy and won't share the data_handle outside?

Reply:
AFAIK, `mem_holder_buffer->get_data_handle()` call does not transfer ownership, it only returns a pointer to the internal data blob. Following line 277 `mem_holder_buffer.reset(new memory(new_buffer_desc, eng));`, mem_holder_buffer maintains the lifespan of the internal memory block. Which means when mem_holder_buffer object is destroyed, the memory pointer returned by the `mem_holder_buffer->get_data_handle()` expires.

Reply:
This memory sharing contributes 2~3% performance improvement with Dien model. 
Basically it is a data buffer. We just need to pass it outside and let "to" memory object share it directly.
A decoupling abstraction layer probably is asked here.

Reply:
defer it as it contributes a little bit but asks much effort.